CardScan/Classes/CreditCardOcr/CreditCardOcrPrediction.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  CreditCardOcrPrediction.swift
+//  ocr-playground-ios
+//
+//  Created by Sam King on 3/19/20.
+//  Copyright © 2020 Sam King. All rights reserved.
+//
+import Foundation
+
+struct CreditCardOcrPrediction {
+    let image: CGImage
+    let number: String?
+    let expiryMonth: String?
+    let expiryYear: String?
+    let name: String?
+    let computationTime: Double
+    let numberBoxes: [CGRect]?
+    let expiryBoxes: [CGRect]?
+    let nameBoxes: [CGRect]?
+    
+    static func emptyPrediction(cgImage: CGImage) -> CreditCardOcrPrediction {
+        CreditCardOcrPrediction(image: cgImage, number: nil, expiryMonth: nil, expiryYear: nil, name: nil, computationTime: 0.0, numberBoxes: nil, expiryBoxes: nil, nameBoxes: nil)
+    }
+    
+    var expiryForDisplay: String? {
+        guard let month = expiryMonth, let year = expiryYear else { return nil }
+        return "\(month)/\(year)"
+    }
+    
+    var expiryAsUInt: (UInt, UInt)? {
+        guard let month = expiryMonth.flatMap({ UInt($0) }) else { return nil }
+        guard let year = expiryYear.flatMap({ UInt($0) }) else { return nil }
+        
+        return (month, year)
+    }
+    
+    var numberBox: CGRect? {
+        let xmin = numberBoxes?.map { $0.minX }.min() ?? 0.0
+        let xmax = numberBoxes?.map { $0.maxX }.max() ?? 0.0
+        let ymin = numberBoxes?.map { $0.minY }.min() ?? 0.0
+        let ymax = numberBoxes?.map { $0.maxY }.max() ?? 0.0
+        return CGRect(x: xmin, y: ymin, width: (xmax - xmin), height: (ymax - ymin))
+    }
+    
+    var expiryBox: CGRect? {
+        return expiryBoxes.flatMap { $0.first }
+    }
+    
+    static func likelyExpiry(_ string: String) -> (String, String)? {
+        guard let regex = try? NSRegularExpression(pattern: "^.*(0[1-9]|1[0-2])\\/([1-2][0-9])$") else {
+            return nil
+        }
+
+        let result = regex.matches(in: string, range: NSRange(string.startIndex..., in: string))
+        
+        if result.count == 0 {
+            return nil
+        }
+        
+        let range1 = Range(result.first!.range(at: 1), in: string)!
```

Comment:
I know `result.first` is taken care of by the previous `result.count` conditional statement but for the range would it be safer to use a default value? Even though we know result will have something but the `!` gives me nightmares.

I'm not sure though.

Reply:
yikes! I'll fix that, nice catch.